kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/KubernetesResourceUtil.java

```diff
@@ -374,4 +380,31 @@ public static Duration getAge(HasMetadata kubernetesResource) {
       return CustomResourceList.class;
     }
   }
+
+  /**
+   * Create Secret by using username and password.
+   * @param dockerServer User to store key value pair for auths map
+   * @param username username that needs to be used during secret creation
+   * @param password password that needs to be used during secret creation
+   * @return an object of Secret
+   */
+  public static Secret createDockerRegistrySecret(String dockerServer, String username, String password) throws JsonProcessingException {
+    Map<String, Object> dockerConfigMap = new HashMap<>();
+    Map<String, Object> auths = new HashMap<>();
+    Map<String, Object> credentials = new HashMap<>();
+    credentials.put("username", username);
+    credentials.put("password", password);
+    String usernameAndPasswordAuth = username + ":" + password;
+    credentials.put("auth", Base64.getEncoder().encodeToString(usernameAndPasswordAuth.getBytes()));
```

Comment:
```suggestion
    credentials.put("auth", Base64.getEncoder().encodeToString(usernameAndPasswordAuth.getBytes(StandardCharsets.UTF_8)));
```

Reply:
Modified and added StandardCharsets.UTF_8.